src/card.rs

```diff
@@ -159,42 +159,71 @@ impl<T: Client, const C: usize, const R: usize> apdu_dispatch::App<C, R> for Car
     }
 }
 
-/// Represent RSA key sizes in bits
-///
-/// Used to configure the maximum sizes allowed for key generation and import through
-/// [`Options::rsa_max_import`] and [`Options::rsa_max_gen`]
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
-pub enum RsaKeySizes {
-    /// RSA key with 2048 bit modulus
-    Rsa2048,
-    /// RSA key with 3072 bit modulus
-    Rsa3072,
-    /// RSA key with 4096 bit modulus
-    Rsa4096,
+bitflags! {
+    /// The algorithms that are allowed to be generated or imported.
+    ///
+    /// Used in [`Options`](allowed_generation) and [`Options`](allowed_imports)
+    #[derive(Clone, Copy, Debug)]
+    pub struct AllowedAlgorithms: u32 {
+        /// P256 NIST curve
+        const P_256 = 1;
+        /// P384 NIST curve
+        const P_384 = 1 << 1;
+        /// P521 NIST curve
+        const P_521 = 1 << 2;
+        /// RSA 2048
+        const RSA_2048 = 1 << 3;
+        /// RSA 3072
+        const RSA_3072 = 1 << 4;
+        /// RSA 4096
+        const RSA_4096 = 1 << 5;
+        /// X25519
+        const X_25519 = 1 << 6;
+        /// EdDsa25519
+        const ED_25519 = 1 << 7;
+        /// BRAINPOOL_P256R1 Brainpool curve
+        const BRAINPOOL_P256R1 = 1<<8;
```

Comment:
tiny nit - `cargo fmt` doesn't catch this ?

```suggestion
        const BRAINPOOL_P256R1 = 1 << 8;
```

Reply:
I suppose no because of the `bitflags` macro.